### PR TITLE
Dilpreet | Test | Integrate bill to bill for export in ledger entry

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/2khener

### CODE REVIEW

**Review by the-prateek-mishra — APPROVED** [review]

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export class ExportLedgerComponent implements OnInit, OnDestroy {
     public emailTypeMini: string = '';
     public emailTypeDetail: string;
     public emailTypeColumnar: string;
+    public emailTypeBillToBill: string;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts]: add js doc

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html [hunk 2]
```diff
@@ -134,7 +156,14 @@ <h3 class="unbold" mat-dialog-title>{{ localeData?.page_heading }}</h3>
                         </mat-slide-toggle>
                     </label>
                 </div>
-                <div class="pd-b2 clearfix" [hidden]="emailTypeSelected !== 'columnar'">
+                <div
+                    class="pd-b2 clearfix"
+                    [hidden]="
+                        emailTypeSelected === 'detailed' ||
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.html]: fix condition

File: apps/web-giddh/src/app/downloads/components/downloads/downloads.component.ts [hunk 3]
```diff
@@ -249,4 +248,16 @@ export class DownloadsComponent implements OnInit, OnDestroy {
         this.destroyed$.complete();
         document.querySelector('body')?.classList?.remove('download-page');
     }
+
+    /**
+   * Callback for translation response complete
+   *
+   * @param {boolean} event
+   * @memberof CustomFieldsListComponent
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/downloads/components/downloads/downloads.component.ts]: fix js doc

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts [hunk 4]
```diff
@@ -77,12 +78,25 @@ export class ExportLedgerComponent implements OnInit, OnDestroy {
...
+    /** This will show/hide for v2 for bill to bill*/
+    public enableBillToBill: boolean = false;
 
     constructor(private ledgerService: LedgerService, private toaster: ToasterService, private permissionDataService: PermissionDataService, private store: Store<AppState>, private generalService: GeneralService, @Inject(MAT_DIALOG_DATA) public inputData, public dialogRef: MatDialogRef<any>, private changeDetectorRef: ChangeDetectorRef, private modalService: BsModalService, private router: Router) {
         this.universalDate$ = this.store.pipe(select(p => p.session.applicationDate), takeUntil(this.destroyed$));
     }
 
     public ngOnInit() {
+        this.voucherApiVersion = this.generalService.voucherApiVersion;
+        this.store.pipe(select(p => p.ledger.account), takeUntil(this.destroyed$)).subscribe(ledgerAccount => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts]: avoid single character variable names

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts [hunk 5]
```diff
@@ -146,35 +161,46 @@ export class ExportLedgerComponent implements OnInit, OnDestroy {
...
-                    if (this.emailTypeSelected === 'admin-detailed') {
-                        if (response.body.encodedData) {
-                            let blob = this.generalService.base64ToBlob(response.body.encodedData, (response.body.type === "xlsx" ? 'application/vnd.ms-excel' : 'text/csv'), 512);
-                            return download(response.body.name, blob, (response.body.type === "xlsx" ? 'application/vnd.ms-excel' : 'text/csv'));
+        if (this.voucherApiVersion === 2 && this.emailTypeSelected === 'billToBill') {
+            this.ledgerService.exportBillToBillLedger(exportRequest, this.inputData?.accountUniqueName).pipe(takeUntil(this.destroyed$)).subscribe(response => {
+                if (response.status === "success") {
+                    let blob = this.generalService.base64ToBlob(response.body.file, 'application/vnd.ms-excel', 512);
+                    return download(`${this.inputData?.accountUniqueName}-bill-to-bill.xlsx`, blob, 'application/vnd.ms-excel');
+                } else if (response.message) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts]:
```suggestion
                } else if (response?.body?.message) {
```

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts [hunk 6]
```diff
@@ -146,35 +161,46 @@ export class ExportLedgerComponent implements OnInit, OnDestroy {
...
-        this.ledgerService.ExportLedger(exportRequest, this.inputData?.accountUniqueName, body.dataToSend, exportByInvoiceNumber).pipe(takeUntil(this.destroyed$)).subscribe(response => {
-            if (response.status === 'success') {
-                if (response.body) {
-                    if (this.emailTypeSelected === 'admin-detailed') {
-                        if (response.body.encodedData) {
-                            let blob = this.generalService.base64ToBlob(response.body.encodedData, (response.body.type === "xlsx" ? 'application/vnd.ms-excel' : 'text/csv'), 512);
-                            return download(response.body.name, blob, (response.body.type === "xlsx" ? 'application/vnd.ms-excel' : 'text/csv'));
+        if (this.voucherApiVersion === 2 && this.emailTypeSelected === 'billToBill') {
+            this.ledgerService.exportBillToBillLedger(exportRequest, this.inputData?.accountUniqueName).pipe(takeUntil(this.destroyed$)).subscribe(response => {
+                if (response.status === "success") {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts]:
```suggestion
                if (response?.status === "success") {
```

File: apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts [hunk 7]
```diff
@@ -146,35 +161,46 @@ export class ExportLedgerComponent implements OnInit, OnDestroy {
...
+                    let blob = this.generalService.base64ToBlob(response.body.file, 'application/vnd.ms-excel', 512);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/material-ledger/components/export-ledger/export-ledger.component.ts]:
```suggestion
                    let blob = this.generalService.base64ToBlob(response?.body?.file, 'application/vnd.ms-excel', 512);
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]
fix changes

**Comment by dilpreetsinghofficial** [comment]
PR reviewed done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**